sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -2336,25 +2336,19 @@ block_store(AccessorT acc, detail::DeviceAccessorOffsetT byte_offset,
           DefaultLSCAlignment);
   constexpr bool AlignmentRequiresLSC =
       PropertyListT::template has_property<alignment_key>() && Alignment < 16;
+  using Tx = detail::__raw_t<T>;
+  constexpr unsigned Sz = sizeof(Tx) * N;
   if constexpr (detail::has_cache_hints<PropertyListT>() ||
-                AlignmentRequiresLSC) {
+                AlignmentRequiresLSC || Sz < detail::OperandSize::OWORD ||
+                Sz % detail::OperandSize::OWORD != 0 ||
+                !detail::isPowerOf2(Sz / detail::OperandSize::OWORD) ||
+                Sz > 8 * detail::OperandSize::OWORD) {
```

Comment:
Minor suggestion. It could be just a bit more readable if it was a separate bool var here, e.g.:
```suggestion
                AlignmentRequiresLSC || SizeRequiresLSC) {
```

Reply:
good point, will do

Reply:
fixed in https://github.com/intel/llvm/pull/13150/commits/fd98ab55c2b3ea4b131cd9cc140e921953228767, thx